### PR TITLE
Feat: flexible csv column names

### PR DESCRIPTION
**Pull request by svenruf**
Allows the user to rename csv columns directly while using the e.g. read_triplegs_csv functions
modified tests accordingly
added test data with modified column names

### CODE REVIEW

File: trackintel/io/file.py [hunk 1]
```diff
@@ -10,6 +10,9 @@ def read_positionfixes_csv(*args, **kwargs):
     builds a geopandas GeoDataFrame. This also validates that the ingested data
     conforms to the trackintel understanding of positionfixes (see 
     :doc:`/modules/model`). 
+    
+    In addition to the pandas read_csv keyword arguments, this function provides to rename columns directly while importing
+    by specifying the columnnames in a dictionary, see second example.
```
Comment (hong2223) [inline trackintel/io/file.py:15]: I am wondering if we need to explicitly write out the `column` as a parameter and describe how to use it (mainly just original col as key and standard col names as value), an example of comment see [here ](https://github.com/mie-lab/trackintel/blob/master/trackintel/preprocessing/positionfixes.py).

File: trackintel/io/file.py [hunk 2]
```diff
@@ -58,12 +64,22 @@ def read_triplegs_csv(*args, **kwargs):
...
+    by specifying the columnnames in a dictionary, see example.
+    
     Returns
     -------
     GeoDataFrame
         A GeoDataFrame containing the triplegs.
+        
+    Examples
+    --------
+    >>> trackintel.read_triplegs_csv('data.csv', columns={'start_time':'started_at'})
```
Comment (hong2223) [inline trackintel/io/file.py:78]: maybe add an example without columns here.

File: tests/io/test_file.py [hunk 3]
```diff
@@ -61,13 +70,17 @@ def test_staypoints_from_to_postgis(self):
...
+        mod_file = os.path.join('tests','data','locations_mod_columns.csv')
         tmp_file = os.path.join('tests', 'data', 'locations_test.csv')
-        locs = ti.read_locations_csv(orig_file, sep=';')
-        locs.as_locations.to_csv(tmp_file, sep=';',
-                                 columns=['user_id', 'elevation', 'center', 'extent'])
+        mod_plcs = ti.read_locations_csv(mod_file, columns={'geom':'center'},sep=';')
+        plcs = ti.read_locations_csv(orig_file, sep=';')
+        assert mod_plcs.equals(plcs)
+        plcs.as_locations.to_csv(tmp_file, sep=';',
+                              columns=['user_id', 'elevation', 'center', 'extent'])
```
Comment (hong2223) [inline tests/io/test_file.py:79]: change the `plcs` (places) to `locs` (locations). We have changed the wording "places" into "locations".

File: trackintel/io/file.py [hunk 4]
```diff
@@ -58,12 +64,22 @@ def read_triplegs_csv(*args, **kwargs):
     builds a geopandas GeoDataFrame. This also validates that the ingested data
     conforms to the trackintel understanding of triplegs (see :doc:`/modules/model`).
 
+
+    In addition to the pandas read_csv keyword arguments, this function provides to rename columns directly while importing
+    by specifying the columnnames in a dictionary, see example.
+
```
Comment (hong2223) [inline trackintel/io/file.py:70]: Comment explicitly, see above

File: trackintel/io/file.py [hunk 5]
```diff
@@ -94,13 +111,18 @@ def read_staypoints_csv(*args, **kwargs):
...
     the ingested data conforms to the trackintel understanding of staypoints 
     (see :doc:`/modules/model`).
+    
+    In addition to the pandas read_csv keyword arguments, this function provides to rename columns directly while importing
+    by specifying the columnnames in a dictionary, see examples in read_triplegs_csv or read_positionfixes_csv.
 
     Returns
     -------
     GeoDataFrame
         A GeoDataFrame containing the staypoints.
```
Comment (hong2223) [inline trackintel/io/file.py:121]: Maybe also add the two examples here

File: trackintel/io/file.py [hunk 6]
```diff
@@ -132,13 +154,17 @@ def read_locations_csv(*args, **kwargs):
     validates that the ingested data conforms to the trackintel understanding 
     of locations (see :doc:`/modules/model`).
 
+    In addition to the pandas read_csv keyword arguments, this function provides to rename columns directly while importing
+    by specifying the columnnames in a dictionary, see examples in read_triplegs_csv or read_positionfixes_csv.
+
```
Comment (hong2223) [inline trackintel/io/file.py:159]: Comment explicitly, see above

File: trackintel/io/file.py [hunk 7]
```diff
@@ -132,13 +154,17 @@ def read_locations_csv(*args, **kwargs):
...
     Returns
     -------
     GeoDataFrame
         A GeoDataFrame containing the locations.
     """
```
Comment (hong2223) [inline trackintel/io/file.py:164]: Maybe also add the two examples here

File: trackintel/io/file.py [hunk 8]
```diff
@@ -170,19 +196,25 @@ def read_trips_csv(*args, **kwargs):
     """Wraps the pandas read_csv function and extraces proper datetimes. This also 
     validates that the ingested data conforms to the trackintel understanding 
     of trips (see :doc:`/modules/model`).
+    
+    In addition to the pandas read_csv keyword arguments, this function provides to rename columns directly while importing
+    by specifying the columnnames in a dictionary, see examples in read_triplegs_csv or read_positionfixes_csv.
```
Comment (hong2223) [inline trackintel/io/file.py:202]: Comment explicitly, see above

File: trackintel/io/file.py [hunk 9]
```diff
@@ -170,19 +196,25 @@ def read_trips_csv(*args, **kwargs):
...
 
     Returns
     -------
     DataFrame
         A DataFrame containing the trips.
     """
```
Comment (hong2223) [inline trackintel/io/file.py:207]: Maybe also add the two examples here

**Review by hong2223 — CHANGES_REQUESTED** [review]
Nice functionality! I am mostly concerned with these issues:
- Comment explicitly about the usage of `column`.
- Complete the example line of code in the comments.
- Change variable name `plcs` to `locs`.

Some of them are legacy issues; it will be nice if you can solve them

File: trackintel/io/file.py (hunk 1 above)
Comment (svenruf) [reply trackintel/io/file.py:15]: Yes I thought about it too, but was not sure if this is "clean" to explicitly describe keyword arguments. But as this is the only one that is used by our function, it makes sense to add this.

**Comment by svenruf** [comment]
included the reviews in the last commit

**Pull request merged**